core/src/main/java/nu/yona/server/goals/service/GoalDto.java

```diff
@@ -61,6 +61,30 @@ protected GoalDto(Optional<LocalDateTime> creationTime)
 		this(null, creationTime, Optional.empty(), null, false /* ignored */);
 	}
 
+	@Override
+	public int hashCode()
+	{
+		return id.hashCode();
+	}
+
+	@Override
+	public boolean equals(Object obj)
+	{
+		if (this == obj)
+			return true;
+		if (getClass() != obj.getClass())
+			return false;
+		GoalDto other = (GoalDto) obj;
+		if (id == null)
+		{
+			if (other.id != null)
```

Comment:
It's Eclipse-generated. I didn't give it any thought (yet).

But this would break the logic, as it would return ``true`` without comparing the IDs.

I've simplified the implementation.